src/main/java/teammates/ui/controller/AdminInstructorAccountAddAction.java

```diff
@@ -272,11 +272,15 @@ private String generateDemoCourseId(String instructorEmail) {
     * @return the first proposed course id. eg.lebron@gmail.com -> lebron.gma-demo
     */
     private String getDemoCourseIdRoot(String instructorEmail) {
-        final String[] splitedEmail = instructorEmail.split("@");
-        final String head = splitedEmail[0];
-        final String emailAbbreviation = splitedEmail[1].substring(0, 3);
-        return head + "." + emailAbbreviation
-                + "-demo";
+        final String[] emailSplit = instructorEmail.split("@");
+        
+        final String username = emailSplit[0];
+        final String host = emailSplit[1];
+        
+        final String head = StringHelper.replaceIllegalChars(username, FieldValidator.REGEX_COURSE_ID, '_');
+        final String hostAbbreviation = host.substring(0, 3);
```

Comment:
Strange request, but can you remove all the `final` keywords? We don't really use `final` keywords for variables in our code base 😅